optimizer_wrapper.rb

```diff
@@ -777,7 +777,7 @@ def self.apply_zones(vrp)
   end
 
   def self.clique_cluster(vrp, cluster_threshold, force_cluster)
-    zip_condition = vrp.matrices.any? && (cluster_threshold.to_f.positive? || force_cluster) && !vrp.schedule?
+    zip_condition = vrp.matrices.any? && (cluster_threshold || force_cluster) && !vrp.schedule?
 
     if zip_condition
       if vrp.services.any?{ |s| s.activities.any? }
```

Comment:
Right, we used to have something for shipments and we removed it in https://github.com/Mapotempo/optimizer-api/pull/240/files#diff-6b6fec5242129149dd15ae48b6f6e8f383e2b202c6e9bc35357552bd2d9b4eb2R780 so I guess it just disappeared when rebasing. 

I think I had made a similar comment by the time, but anyways. 

Just to make sure we are one the same page, you suggest that we raise as soon as one service needs to be merged with another one, and they are both included in a (different) relation which type is in : 
* same_vehicle, 
* same_route, 
* sequence,
* order
* shipment 
* meetup
Is that right ?

Reply:
I skimmed the zip_cluster code and didn't see any logic to handle the relations of merged services (can you verify please in case I missed something).

Yeah similar to that and I think raise is more aptly (similar to how it is done vis-a-vis activities, just below the code you shared) because otherwise we silently ignore the threshold setting.

My suggestion was more rudimentary (and easy) -- like it is already done vis-a-vis service.activities, we don't verify if a services with activities is getting merged or not, we just refuse cluster_threshold option -- i.e., if zip_condition is true but we have `vrp.relations.any?{ |r| RELATIONS_ZIP_CLUSTER_CAN_HANDLE.exclude?(r.type) }` then we raise `clsuter threshold option is compatible only with the following relations: #{RELATIONS_ZIP_CLUSTER_CAN_HANDLE}` or something like that.

Reply:
And `RELATIONS_ZIP_CLUSTER_CAN_HANDLE` is all the relations that doesn't get affected by merging services (i.e., the ones that only effect vehicles). This way, when we add a new relation types, the refusal will be handled automatically.